Tools/ardupilotwaf/configuration_helper.py

```diff
@@ -0,0 +1,28 @@
+"""
+WAF Tool that helps on configuration parameter, creating the AP_Config
+header file.
+
+This tool needs compiler_c and compiler_cxx to be loaded, make sure
+you load them before this tool.
+
+Example::
+    def configure(cfg):
+        cfg.load('configuration_helper')
+"""
+
+def configure(cfg):
+    cfg.check_cxx(fragment='#include<cmath>\n int main() { std::isfinite(1); }\n',
+                  define_name="HAVE_CMATH_ISFINITE", mandatory=False)
+
+    cfg.check_cxx(fragment='#include<math.h>\n int main() { isinf(1); }\n',
+                  define_name="HAVE_MATH_ISINF", mandatory=False)
+
+    cfg.check_cxx(fragment='#include<cmath>\n int main() { std::isinf(1); }\n',
+                  define_name="HAVE_CMATH_ISINF", mandatory=False)
+
+    cfg.check_cxx(fragment='#include<math.h>\n int main() { isnan(1); }\n',
+                  define_name="HAVE_MATH_ISNAN", mandatory=False)
+
+    cfg.check_cxx(fragment='#include<cmath>\n int main() { std::isnan(1); }\n',
+                  define_name="HAVE_CMATH_ISNAN", mandatory=False)
```

Comment:
Some comments:
 - Those checks should be done for float literals, i.e., `1.0f`. Otherwise, the checks for `HAVE_MATH_*` will always fail. The way this is written, the check `#if defined(HAVE_CMATH_*) && !defined(HAVE_MATH_*)` will always be true as long as `defined(HAVE_CMATH_*)` is true. Maybe that's why this has (supposably) run successfully in your environment. If using a float literal instead of `1`, you'll probably experience an error in you environment because of the my next comment.
 - I'm almost sure those tests don't cover the case we ran into, i.e., including a header that includes `cmath` direct or indirectly, which somehow makes `math.h`'s macros undefined. For example:
```c++
#include <algorithm> // this header
#include <math.h>
...
```